network/peer/upgrader_test.go

```diff
@@ -0,0 +1,222 @@
+				MinVersion:         tls.VersionTLS13,
+				Certificates:       []tls.Certificate{testCase.genClientTLSCert()},
+			}
+
+			listener, err := net.Listen("tcp", "127.0.0.1:0")
+			require.NoError(t, err)
+			defer listener.Close()
+
+			var wg sync.WaitGroup
+			wg.Add(1)
+
+			go func() {
+				defer wg.Done()
+				conn, err := listener.Accept()
+				require.NoError(t, err)
+
+				_, _, _, err = upgrader.Upgrade(conn)
+
+				require.ErrorIs(t, err, testCase.expectedErr)
+			}()
+
+			conn, err := tls.Dial("tcp", listener.Addr().String(), &clientConfig)
+			require.NoError(t, err)
+
+			require.NoError(t, conn.Handshake())
+
+			wg.Wait()
+		})
+	}
+}
+
+func nonStandardRSAKey(t *testing.T) *rsa.PrivateKey {
+	for {
+		sk, err := rsa.GenerateKey(rand.Reader, 2048)
+		require.NoError(t, err)
+
+		// This speeds up RSA operations, and was initialized during the key-gen.
+		// If we wish to override the key parameters we need to nullify this,
+		// otherwise the signer will use these values and the verifier will use
+		// the values we override, and verification will fail.
+		sk.Precomputed = rsa.PrecomputedValues{}
+
+		// We want a non-standard E, so let's use E = 257 and derive D again.
+		e := 257
+		sk.PublicKey.E = e
+		sk.E = e
+
+		p := sk.Primes[0]
+		q := sk.Primes[1]
+
+		pminus1 := new(big.Int).Sub(p, big.NewInt(1))
+		qminus1 := new(big.Int).Sub(q, big.NewInt(1))
+
+		phiN := big.NewInt(0).Mul(pminus1, qminus1)
+
+		sk.D = big.NewInt(0).ModInverse(big.NewInt(int64(e)), phiN)
+
+		if sk.D == nil {
+			// If we ended up picking a bad starting modulus, try again.
+			continue
+		}
+
+		return sk
+	}
+}
+
+func makeTLSCert(t *testing.T, privKey *rsa.PrivateKey) tls.Certificate {
+	x509Cert := makeRSACertAndKey(t, privKey)
+
+	rawX509PEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: x509Cert.cert.Raw})
+	privateKeyInDER, err := x509.MarshalPKCS8PrivateKey(x509Cert.key)
+	require.NoError(t, err)
+
+	privateKeyInPEM := pem.EncodeToMemory(&pem.Block{Type: "PRIVATE KEY", Bytes: privateKeyInDER})
+
+	tlsCertServer, err := tls.X509KeyPair(rawX509PEM, privateKeyInPEM)
+	require.NoError(t, err)
+
+	return tlsCertServer
+}
+
+type certAndKey struct {
+	cert x509.Certificate
+	key  *rsa.PrivateKey
+}
+
+func makeRSACertAndKey(t *testing.T, privKey *rsa.PrivateKey) certAndKey {
+	// Create a self-signed cert
+	basicCert := basicCert()
+	certBytes, err := x509.CreateCertificate(rand.Reader, basicCert, basicCert, &privKey.PublicKey, privKey)
+	require.NoError(t, err)
+
+	cert, err := x509.ParseCertificate(certBytes)
+	require.NoError(t, err)
+
+	return certAndKey{
+		cert: *cert,
+		key:  privKey,
+	}
+}
+
+func basicCert() *x509.Certificate {
+	return &x509.Certificate{
+		SerialNumber:          big.NewInt(0).SetInt64(100),
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(time.Hour).UTC(),
+		BasicConstraintsValid: true,
+	}
+}
+
+type mockPrometheusCounter struct {
+	t *testing.T
```

Comment:
`t` never seems to be used
```suggestion
```

Reply:
[Fixed](https://github.com/ava-labs/avalanchego/pull/3501/commits/e45a72ae90a2c993915320789a299e78557fccac#diff-904154d6e72be15a120d196a65b860c273801bb294ffca48bdf08e377a5f33a3L215)